### PR TITLE
Merge system-trace into core

### LINKED ISSUE #979: merge system-trace into core
let merge the system-trace extension into core

- [ ]  merge into core
- [ ]  add and check tests
- [ ] rewrite grunt task

### PR DESCRIPTION
**Pull request by m-mujica**
Closes #979 

the system-trace tests pass on the browser but I'm getting a weird error (the system-trace test does load `@loader` but not sure why it will failed to load) 

```
Warning: Cannot find module '@loader' Use --force to continue.
Error: Cannot find module '@loader'
  at Function.Module._resolveFilename (module.js:469:15)
  at Function.Module._load (module.js:417:25)
  at Module.require (module.js:497:17)
```

I've no idea why that's happening.... Rings any 🔔 @matthewp?

@pYr0x I hope you don't mind me working on this issue =)

### CODE REVIEW

**Comment by pYr0x** [comment]
no problem @m-mujica .
i talked with matthew and we want to put all extensions and tests in there own folder like https://github.com/stealjs/steal/tree/f8114d03f4ee4e085f87cd746fdc1c9130238726/src/cache-bust

File: src/system-trace.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+	loader.eachModule = function(cb){
+		for (var moduleName in this._loader.modules) {
+			cb.call(this, moduleName, this.get(moduleName));
+		}
+	};
+}
+
+if(typeof System !== "undefined") {
+	applyTraceExtension(System);
```
Comment (pYr0x) [inline src/system-trace.js]: we have that new `addStealExtension` method that should be called if adding a new extension to steal
for example, see
https://github.com/stealjs/steal/blob/f8114d03f4ee4e085f87cd746fdc1c9130238726/src/cache-bust/cache-bust.js#L4`

**Comment by m-mujica** [comment]
screen shot 2017-01-24 at 09 03 28

**Comment by matthewp** [comment]
@m-mujica can you rename to just `trace`? Nothing system about it :)

**Comment by m-mujica** [comment]
@matthewp renamed!

**Pull request merged**